src/stream/DOT.hpp

```diff
@@ -55,22 +55,14 @@ class DOT : public KernelBase
 
   void setCudaTuningDefinitions(VariantID vid);
   void setHipTuningDefinitions(VariantID vid);
-  template < size_t block_size >
-  void runCudaVariantBlockAtomic(VariantID vid);
-  template < size_t block_size >
-  void runHipVariantBlockAtomic(VariantID vid);
-  template < size_t block_size >
-  void runCudaVariantBlockAtomicOccGS(VariantID vid);
-  template < size_t block_size >
-  void runHipVariantBlockAtomicOccGS(VariantID vid);
-  template < size_t block_size >
-  void runCudaVariantBlockDevice(VariantID vid);
-  template < size_t block_size >
-  void runHipVariantBlockDevice(VariantID vid);
-  template < size_t block_size >
-  void runCudaVariantBlockDeviceOccGS(VariantID vid);
-  template < size_t block_size >
-  void runHipVariantBlockDeviceOccGS(VariantID vid);
+  template < size_t block_size, bool direct >
```

Comment:
👍

Reply:
I was debating whether I should pass the actual class through instead of just a bool to increase type safety. That way you'd have to do something like this `Direct::direct` which would only work if you passed in the right type.

Reply:
fixed